### ISSUE TITLE
DOS 4.4 - assert during INIT operation

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#533]
In this function (debug build) `FormatTrack::DecodeLatchNibble()`, this line asserts:
`_ASSERT( (m_bmWrittenSectorAddrFields & (1<<sector)) == 0 );`

because:
* m_bmWrittenSectorAddrFields	= 0x03ff
* sector = 0

Ignoring the asserts, INIT fails with:
ERR=$00
ERR=$08

---

FYI, DOS 4.4 info here:

http://www.brutaldeluxe.fr/projects/dos44/index.html

> The DOS 4.4 encoding scheme is called 4*4 and requires 512 nibbles to store 256 bytes onto disk, we are really wasteful!
> As the number of bits we can write per track is limited, we have limited DOS 4.4 to 10 sectors per track

NB. There's no checksum after the 512 4&4 data, so the data field is 512 nibbles (not 513).

**Comment by tomcw** [comment]
This is not a regression - same errors for:
* 1.26.3.6 (for both authentic & enhanced disk access speed) - this version didn't have the improved DOS 3.3/ProDOS formatter support
* 1.25.0.4 (for both authentic & enhanced disk access speed)

NB. INIT'ing with a .nib in drive-2.

**Comment by tomcw** [comment]
DOS 4.4's INIT writes 10 sectors (in the order 0,1,...,9), then reads looking for sector-0.

These 10 sectors are ~0x1709 nibbles, meaning that there's 0x1F6(to end of 0x1A00 .nib track) + 0x80(gap1) nibbles before sector-0 starts.

So the read looking for sector-0 will "timeout" by reading too many nibbles before sector-0. This forces repeated retries and then failure (ERR=$00/$08).

This can be fixed (just like #125), but detecting a track init, and then truncating to a shorter track length.

Specifically for this DOS 4.4 format, count the size for the data field (during a write) and then in `FormatTrack::DriveSwitchedToReadMode()`:

```
	const UINT kDataFieldLength44 = 256*8/4;	// 512 (no data checksum)

	if (m_WriteDataFieldSize == kDataFieldLength44)	// DOS 4.4 (ie. 4&4 encoding for data) (GH#533)
	{
		if (m_bmWrittenSectorAddrFields != 0x03FF || m_WriteDataFieldPrologueCount != 10)	// written all 10 sectors?
			return;
	}
```

**Comment by tomcw** [comment]
DOS 4.4 is just a proof of concept. To date nothing uses it, and it doesn't offer anything (beyond PoC).

If AppleWin were to support it, then it means extra testing and maintenance for a format no one is requesting support for, nor currently of any practical use.

It has been interesting investigating how AppleWin could support it, but no more.

So for now, I will close this issue as WON'T FIX.
